core/src/main/java9/feign/Capability.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright 2012-2021 The Feign Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package feign;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.List;
+import feign.Logger.Level;
+import feign.Request.Options;
+import feign.codec.Decoder;
+import feign.codec.Encoder;
+
+/**
+ * Capabilities expose core feign artifacts to implementations so parts of core can be customized
+ * around the time the client being built.
+ *
+ * For instance, capabilities take the {@link Client}, make changes to it and feed the modified
+ * version back to feign.
+ *
+ * @see Metrics5Capability
+ */
+public interface Capability {
```

Comment:
What is the difference between regular `Capability.java` and java 9 copy of it?!

Reply:
Same goes for every other duplicate class, yes. I'm still working on the testing, which is the hardest part I'm finding out.